Making changes to creating-an-addon.md and creating guide for making-a-weapon.md (#7)
TTT-2/docs: docs/developers/content-creation/creating-a-weapon.md

# Creating a Weapon

When starting content creation for gmod, creating a basic weapon using the SWEP structure is one of the easiest things you can do.

> For this guide I'll be using the Visual Studio Code with the glualint extension as referenced in the "Creating an addon" section of these documents.

Reviewer: You could also link to the 'editor' section of the 'creating an addon' page here. See my comment below for an example

## File Format

Firstly you need to make sure that you're using the correct folder structure.
Your addon folder must be in this format:

> * `weapon_ttt_<weapon name>\gamemodes\terrortown\entities\weapons\weapon_ttt_\<weapon name>.lua`

Reviewer 2: the addon folder doesn't need to have this name, I'd replace this and the following lines with `<addon_name>\gamemodes\terrortown\entities\weapons\weapon_ttt_\<weapon name>.lua`

Also we are using forward slashes in our documentation

Author: Sorry, Windows habits again

Reviewer 2: No need to feel sorry. Your work is great!

Author: Awh many thanks

Your .vmt and .vtf icon files (what gets shown in the buy menu) will need to be stored in this format:

* `\weapon_ttt_<weapon name>\materials\vgui\ttt\<weapon name>_icon.vtf`
* `\weapon_ttt_<weapon name>\materials\vgui\ttt\<weapon name>_icon.vmt`

> See the [Creating An Addon](https://docs.ttt2.neoxult.de/developers/basics/creating-an-addon/) page for more info on folder structure

Reviewer: i would suggest using a relative path instead of an absolute path. Also you can directly link to the exact section you are refering to :) 
```suggestion
See the [Creating An Addon](../../basics/creating-an-addon/#project-structure) page for more info on folder structure.
```

## Important 'Structural' features

> ### If SERVER Block

Reviewer 2: I'd rename this to something more fitting like `General Setup`

```lua
if SERVER then
   -- Shares the specified file with the client (if blank shares current file)
   AddCSLuaFile()

   -- Shares the buy menu icon with the client
   resource.AddFile("materials/vgui/ttt/blue_template_icon.vmt")
end

> ```

Reviewer: Please add an empty line after a code segment

It's very important to create a globally unique name for this because the Gmod
client will look for the given path locally to display it. So if there are two
identical icon names the client may display the incorrect one

### If CLIENT Block

```lua
if CLIENT then
   -- Client viewpoint vals
   SWEP.PrintName          = "Template Weapon"

>    SWEP.Slot               = 6 -- (See SWEP.Kind below)

Reviewer 2: the slot isn't needed in TTT2 as it is derived from the SWEP.Kind

   SWEP.ViewModelFOV       = 70 -- How much of the first person viewmodel is seen
   SWEP.ViewModelFlip      = true -- If true, the viewmodel is on the left
   SWEP.DrawCrossHair      = false -- Does the client draw its own crosshair over the default ones

   -- Client menu data
   SWEP.Icon = "vgui/ttt/blue_template_icon.vtf" -- The filename of the icon shown in the buymenu
   -- The information shown in the buy menu
   SWEP.EquipMenuData      =
   {
      type = "Weapon",

>       desc = "This is a template weapon"

Reviewer 2: I'd reccomend showing how to use translations here. See: 

https://github.com/TTT-2/ttt2-wep_doorlocker/blob/master/gamemodes/terrortown/entities/weapons/weapon_ttt_doorlocker.lua#L21-L25

Author: Ah, this is new to me too - I'm afraid I'll probably finish that off tomorrow

Reviewer 2: No hurry. I just added a new article about language support over here: https://docs.ttt2.neoxult.de/developers/content-creation/language-support/

Author: ta

Author: btw on the docs page you say that each language identifier must be unique, but on your doorlocker they're just the language names. Is this a mistake or have I got language identifier mixed up with the lua file?

Reviewer 2: could you explain what you mean by "not unique"?

Author: Oh I've got mixed up then, because i thought you meant this was the language identifier
![image](https://user-images.githubusercontent.com/67487108/86126450-43b62680-bad6-11ea-8e25-d81b7ced9cfe.png) from the lang support page

Reviewer 2: Ah no, the language itself shouldn't be unique, maybe I have to specify this

Author: ah aight thank you

>    };

Reviewer 2: no semicolons in lua

Author: old habits from java D:

end
```

> 

Reviewer: More than one empty line doesn't do anything. please remove the two extra lines

## Important SWEP values

```lua
-- What weapon this is based upon (always use weapon_tttbase when creating a ttt/2 weapon)
SWEP.Base                  = "weapon_tttbase"

> ```

Reviewer: Please add an empty line after the code segment

The TTT2 gamemode has a file which describes the default values and methods. Any SWEP which specifies weapon_tttbase inherits the attributes and methods unless specified.

```lua
-- Primary fire vals
SWEP.Primary.Delay         = 0.001 -- Higher is slower
SWEP.Primary.Automatic     = true -- Is it automatic
SWEP.Primary.Damage        = 1 -- How much damage per shot
SWEP.Primary.Cone          = 0.5 -- Cone of accuracy
SWEP.Primary.Ammo          = "smg1" -- Which ttt ammo entity is picked up by this swep
SWEP.Primary.ClipSize      = 100 -- Maximum clip size
SWEP.Primary.DefaultClip   = 100 -- Starting clip size
SWEP.Primary.Sound         = Sound("Weapon_AK47.Single") -- Either a sound file from the default list or a custom one
SWEP.Primary.NumShots      = 1 -- Number of shots fired at one time (make into shotgun if increase conde val)
SWEP.Primary.Recoil        = 1.5 -- Recoil value (bigger = bigger)

> ```

Reviewer: Please add an empty line after the code segment

> Note that all of these attributes can also be applied to SWEP.Secondary which is alt click if you disable ironsights

Reviewer: if you want to you could use the following syntax to create something like this: 
![image](https://user-images.githubusercontent.com/16392835/85954177-dd19f700-b975-11ea-8d3f-68a0a716d0f5.png)
```suggestion
!!! note
    All of these attributes can also be applied to SWEP.Secondary which is alt click if you disable ironsights
```

Author: I can't seem to replicate this effect using that syntax? Although I must admit I'm using the built in MD previewer in VSC

Reviewer 2: This is a custom feature of MKDocs. This does not work in VSCode. Just take a look at our other pages and how we've done it there

SWEP.UseHands              = true -- Whether the swep will force the user to see the viewmodel's hands
SWEP.AutoSpawnable         = false -- Whether the swep will spawn upon map gen (override as false if EQUIP1/2)
SWEP.AmmoEnt               = "item_ammo_smg1_ttt" -- Which entity you pick up to fill your clips
SWEP.InLoadoutFor          = nil -- Which roles gain this weapon upon round start (table data structure)
SWEP.LimitedStock          = true -- If true only one can be bought per round
SWEP.AllowDrop             = true -- Is the player able to drop the swep
SWEP.IsSilent              = false -- If true, the killed player will not scream upon death
SWEP.NoSights              = false -- If true, the swep has no ironsights capability (no Secondary fire if ironsights?)
SWEP.HeadshotMultiplier    = 50 -- Multiply the headshot damage by this much
SWEP.DeploySpeed           = 0.2 -- Lower the slower
SWEP.IronSightsPos         = Vector(-2, -5, 0) -- Moves the Viewmodel by this vector when using ironsights
SWEP.IronSightsAng         = Vector(-2, 0, 0) -- Rotates the Viewmodel by this angle when using ironsights

> ```

Reviewer: Please add an empty line after the code segment

These are most of the other miscellaneous values which can also be applied to the SWEP (That I have found, if any others are found to be applicable to TTT2 please notify us). Note that if any of these values are not explicitly stated in your lua file they will inherit the weapon_tttbase ones, or cause errors.

```lua
SWEP.Kind                 = WEAPON_EQUIP1 -- Kind specifies what weapon_ttt category it falls into
--[[
   (        0      1      2       3      4     5      6       7       8   ) SWEP.Slot
   (WEAPON_ MELEE, HEAVY, PISTOL, HEAVY, NADE, CARRY, EQUIP1, EQUIP2, ROLE) SWEP.Kind
--]]

> ```

Reviewer: Please add an empty line after the code segment

+1 to the SWEP.Slot get the eqip slot key

```lua
SWEP.CanBuy                = {ROLE_TRAITOR} -- Which roles can purchase this swep (table)

> ```

Reviewer: Please add an empty line after the code segment

# Creating a Weapon

When starting content creation for gmod, creating a basic weapon using the SWEP structure is one of the easiest things you can do.
For this guide I'll be using the Visual Studio Code with the glualint extension as referenced in the [Creating An Addon](../basics/creating-an-addon.md) section of these documents. Using those instructions for setting up the editor and project structure/management will make it easier for us to support you with any problems while coding.

## File Format

The most important thing you must get right about weapons (or addons in general) is the folder structure. If your structure is all wrong, them gmod won't recognise the files and your addon will break.

### Weapon Lua Script

Your weapon's lua file must be in this format:

* `<addon name>/gamemodes/terrortown/entities/weapons/weapon_ttt_/<weapon name>.lua`

### Icon Files

> Your .vmt and .vtf icon files (what gets shown in the buy menu) will need to be stored in this format:

Reviewer 2: ```suggestion
Your `*.vmt` and `*.vtf` icon files (what gets shown in the buy menu) will need to be stored in this format:
```

* `<addon name>/materials/vgui/ttt/<weapon name>_icon.vtf`
* `<addon name>/materials/vgui/ttt/<weapon name>_icon.vmt`

Documentation on how to create icons will be uploaded soon.

### Model Files

Model files for the SWEP will also need to be stored in the addon folder, unless you're using the default ones which come with source games. It's advised that even if you're using non-GMod models that you include them in the addon because not all users will own CS:S or Half life, etc.

* `<addon name>/materials/models/<model folder>/<model name>.vmt`
* `<addon name>/models/<model files>`

Documentation on how to create models will be uploaded soon.

GMod default weapon model paths are found [here](https://wiki.facepunch.com/gmod/Common_Weapon_Models).

### Sound Files

Custom sounds for your SWEP will need to be stored in this folder if you're not going to use the default sounds from GMod. This has the benefit that they're only loaded when you're using the ttt/2 gamemode.

* `<addon name>/gamemodes/terrortown/content/sound/`

!!! note

>    Sounds types accepted are: .wav, .mp3 and .ogg.

Reviewer 2: ```suggestion
   Sounds types accepted are: `*.wav`, `*.mp3` and `*.ogg`.
```

SWEP.HeadshotMultiplier    = 50 -- Multiply the headshot damage by this much
SWEP.DeploySpeed           = 0.2 -- Lower the slower
SWEP.IronSightsPos         = Vector(-2, -5, 0) -- Moves the Viewmodel by this vector when using ironsights
SWEP.IronSightsAng         = Vector(-2, 0, 0) -- Rotates the Viewmodel by this angle when using ironsights
```

These are most of the other miscellaneous values which can also be applied to the SWEP (That I have found, if any others are found to be applicable to TTT2 please notify us). Note that if any of these values are not explicitly stated in your lua file they will inherit the weapon_tttbase ones, or cause errors.

```lua
SWEP.Kind                 = WEAPON_EQUIP1 -- Kind specifies what weapon_ttt category it falls into
--[[
   (        0      1      2       3      4     5      6       7       8   ) SWEP.Slot
   (WEAPON_ MELEE, HEAVY, PISTOL, HEAVY, NADE, CARRY, EQUIP1, EQUIP2, ROLE) SWEP.Kind
--]]
```

+1 to the SWEP.Slot get the eqip slot key

```lua
SWEP.CanBuy                = {ROLE_TRAITOR} -- Which roles can purchase this swep (table)
```

> This data is stored in a table, its the roles which can purchase it in the buy menu. Convention is that the keyword for each role is ROLE_\<ROLE NAME>.

Reviewer: This line looks like this:
![image](https://user-images.githubusercontent.com/16392835/86133183-44f04f00-bae8-11ea-8369-9169a4050fdf.png)
with mkdocs.

The following renders correctly for me
```suggestion
This data is stored in a table, its the roles which can purchase it in the buy menu. Convention is that the keyword for each role is ROLE_<ROLE NAME\>.
```

Author: thanks, had problems with how mkdocs interpreted the \ escape char :/
